_posts/en/newsletters/2019-01-15-newsletter.md

```diff
@@ -0,0 +1,94 @@
+---
+title: 'Bitcoin Optech Newsletter #29'
+permalink: /en/newsletters/2019/01/15/
+name: 2019-01-15-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter announces a security upgrade for C-Lightning,
+describes a paper and additional research into wallets that accidentally
+revealed their private keys, and lists some notable code changes in
+popular Bitcoin infrastructure projects.
+
+## Action items
+
+- **Upgrade to C-Lightning 0.6.3:** this release fixes a remote DoS
+  vulnerability that could be used to crash C-Lightning nodes and
+  potentially steal money.  See the *notable code changes* section below
+  for details.  This release also includes other less critical bug fixes
+  and new features.
+
+## News
+
+- **Weak signature nonces discovered:** a preprint [paper][weak nonces]
+  by researchers Joachim Breitner and Nadia Heninger describes how they
+  discovered hundreds of Bitcoin private keys by looking for signatures
+  generated using nonces with less than the expected entropy of 256
+  bits.  Independent [code archaeology][gmaxwell bitcore] by Gregory
+  Maxwell indicates that the main culprit was probably the BitPay
+  Bitcore software which introduced a bug around July 2014 and released
+  a fix about a month later.  (Note: BitPay Bitcore is unrelated to
+  Bitcoin Core.)  From there, the bug propagated to software such as
+  BitPay Copay that depended upon Bitcore.  About 97% of the faulty
+  signatures found in the paper are compatible with Maxwell's Copay
+  hypothesis, and the paper provides plausible explanations for most of
+  the remaining 3% of signatures, indicating that users of modern
+  wallets are probably safe provided they do not continue to use
+  addresses whose bitcoins they spent using earlier vulnerable programs.
+
+    If you ever used an affected version of Bitcore (0.1.28 to 0.1.35),
+    Copay (0.4.1 to 0.4.3), or other vulnerable software, you should
+    create a new wallet file, send all of your funds from the old wallet
+    file to an address in the new wallet, and discontinue use of the
+    previous wallet file.  When designing software that signs Bitcoin
+    transactions, you should prefer to use peer-reviewed implementations
+    that generate signature nonces determinstically, such as
+    [libsecp256k1][] which implements [RFC6979][].
+
+## Notable code changes
+
+*Notable code changes this week in Bitcoin Core, LND, C-lightning,
```

Comment:
Consider making these links to the repos:

- https://github.com/bitcoin/bitcoin
- https://github.com/lightningnetwork/lnd/
- https://github.com/ElementsProject/lightning
- https://github.com/ACINQ/eclair
- https://github.com/bitcoin-core/secp256k1